### DISCUSSION TITLE
Validators report STAC as valid although it is not

### DISCUSSION THREAD

**Discussion by m-mohr** [radiantearth/stac-spec#1242] [STAC Software]
The JSON Schemas are not very complete with regards to (mostly) assets, links and summaries, e.g. #1241. Many related requirements are not checked. The Validators that we have all report something similar to a boolean "valid" or "invalid" although they can't fully guarantee the actual file is valid. Thus, implementors claim their implementation is valid although it is not and blame us for that.

In JSON Schema draft-07 we can't easily validate assets anyway due to them being an object of objects. We'd need to update to the upcoming draft and then it's still a major effort to make the JSON Schemas more complete (and they will likely never be fully complete).

So I'm wondering whether we should update the validators to explain that?
So if the validation status is currently "valid", maybe it should just say something like "no obvious errors found (but that doesn't guarantee correctness)".

Thoughts?

**Comment by jonhealy1** [comment]
We could add a disclaimer message to `stac-validator` output but it would be hard to move away from boolean validation. We could use `passed` which is maybe closer to the truth than `valid`? We can't guarantee validity but we can say that an object has `passed` or `failed` jsonschema validation with certainty.

**Reply by m-mohr** [reply]
Yeah, that makes sense. I would be onboard to change this in stac-node-validator and STAC Browser, too.

**Reply by jonhealy1** [reply]
Do you want to write a short disclaimer message we can use? We could also link back to this issue if people have thoughts or questions?

**Reply by m-mohr** [reply]
You are the native speaker, aren't you? :-)
But here's my first proposal:
> Schema-based STAC validation may be incomplete and should only be considered as a first indicator of validity.

Happy to improve it based on your suggestions.

**Reply by m-mohr** [reply]
Maybe we should write up somewhere what the known limitations are and link to that issue. This discussion is probably a bit vague right now.

**Reply by jonhealy1** [reply]
I think that would be really good. The message you wrote is perfect I think.

**Reply by m-mohr** [reply]
Cool, I've opened issues on my repos already:
- https://github.com/stac-utils/stac-node-validator/issues/90
- https://github.com/radiantearth/stac-browser/issues/610

**Reply by jonhealy1** [reply]
Ok looks good, I did the same in stac-check/ validator

- https://github.com/stac-utils/stac-validator/issues/261
- https://github.com/stac-utils/stac-check/issues/133
